tests/extproc/extproc_test.go

```diff
@@ -0,0 +1,162 @@
+//go:build extproc_e2e
+
+package extproc
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/openai/openai-go"
+	"github.com/openai/openai-go/option"
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/yaml"
+
+	"github.com/envoyproxy/ai-gateway/extprocconfig"
+)
+
+//go:embed envoy.yaml
+var envoyYamlBase string
+
+var (
+	openAISchema     = extprocconfig.VersionedAPISchema{Schema: extprocconfig.APISchemaOpenAI}
+	awsBedrockSchema = extprocconfig.VersionedAPISchema{Schema: extprocconfig.APISchemaAWSBedrock}
+)
+
+// TestE2E tests the end-to-end flow of the external processor with Envoy.
+//
+// This requires the following environment variables to be set:
+//   - TEST_AWS_ACCESS_KEY_ID
+//   - TEST_AWS_SECRET_ACCESS_KEY
+//   - TEST_OPENAI_API_KEY
+//
+// The test will be skipped if any of these are not set.
+func TestE2E(t *testing.T) {
+	requireBinaries(t)
+	requireRunEnvoy(t)
+	configPath := t.TempDir() + "/extproc-config.yaml"
+	requireWriteExtProcConfig(t, configPath, &extprocconfig.Config{
+		InputSchema: openAISchema,
+		// This can be any header key, but it must match the envoy.yaml routing configuration.
+		BackendRoutingHeaderKey: "x-selected-backend-name",
+		ModelNameHeaderKey:      "x-model-name",
+		Rules: []extprocconfig.RouteRule{
+			{
+				Backends: []extprocconfig.Backend{{Name: "openai", OutputSchema: openAISchema}},
+				Headers:  []extprocconfig.HeaderMatch{{Name: "x-model-name", Value: "gpt-4o-mini"}},
+			},
+			{
+				Backends: []extprocconfig.Backend{{Name: "aws-bedrock", OutputSchema: awsBedrockSchema}},
+				Headers:  []extprocconfig.HeaderMatch{{Name: "x-model-name", Value: "us.meta.llama3-2-1b-instruct-v1:0"}},
+			},
+		},
+	})
+	requireExtProc(t, configPath)
+
+	client := openai.NewClient(option.WithBaseURL("http://localhost:1062/v1/"))
+	for _, modelName := range []string{
+		"gpt-4o-mini", // This will go to "openai"
+		// TODO: enable after we migrate to sign requests in extproc.
+		//"us.meta.llama3-2-1b-instruct-v1:0", // This will go to "aws-bedrock".
```

Comment:
this is commented out as per the offline discussion cc @yuzisun - i will re-enable after adding the signing code